libres/lib/analysis/update.cpp

```diff
@@ -711,7 +707,7 @@ bool smoother_update(std::vector<int> step_list,
                 obs_data_allocE(obs_data, shared_rng, active_ens_size);
 
             // Part 1: Parameters which do not have row scaling attached.
-            auto A =
+            matrix_type *A =
```

Comment:
According to [Eigen's "Common pitfalls" section](https://eigen.tuxfamily.org/dox/TopicPitfalls.html), `auto` should not be used.